tests_and_analysis/test/script_tests/test_dos.py

```diff
@@ -79,16 +81,36 @@ def test_invalid_file_raises_value_error(self, dos_args):
         with pytest.raises(ValueError):
             euphonic.cli.dos.main(dos_args)
 
+    def test_adaptive_and_lorentz_raises_value_error(self):
+        with pytest.raises(ValueError):
+            euphonic.cli.dos.main([quartz_fc_file, '--adaptive',
+                                   '--shape', 'lorentz'])
+
 
 @patch('matplotlib.pyplot.show')
 @pytest.mark.skip(reason='Only run if you want to regenerate the test data')
 def test_regenerate_dos_data(_):
-    json_data = {}
+    # Read from existing file first to allow option of only replacing for
+    # certain test cases or keys
```

Comment:
Presumably that has to be accompanied by commenting out a bunch of cases? Still, seems a good idea! Better than picking through the JSON diff to find the _intended_ changes.